lib/fs-hash-diff.js

```diff
@@ -0,0 +1,77 @@
+/// @ts-check
+"use strict";
+
+// Imported for type annotations.
+const Entry = require("fs-tree-diff/lib/entry").default; // jshint ignore:line
+const FSTree = require("fs-tree-diff");
+
+class HashEntry {
+  /**
+   * @param relativePath {string}
+   * @param hash {string}
+   */
+  constructor(relativePath, hash) {
+    this.relativePath = relativePath;
+    this.hash = hash;
+  }
+
+  isDirectory() {
+    return false;
+  }
+
+  /**
+   * Whether the entries have the same content.
+   * @param other {unknown}
+   * @returns
+   */
+  equals(other) {
+    if (other instanceof HashEntry) {
+      return this.hash === other.hash;
+    } else {
+      return false;
+    }
+  }
+}
+
+class FSHashTree extends FSTree {
+  /**
+   * Creates an instance of FSHashTree.
+   * @param [options] {{entries?: Array<Entry|HashEntry>, sortAndExpand?: boolean}}
+   */
+  constructor(options) {
+    super(options);
+    /** @type Array<Entry|HashEntry> */
+    this.entries = options.entries || [];
+  }
+  /**
+   * @param entryA {HashEntry | Entry}
+   * @param entryB {HashEntry | Entry}
+   * @returns {boolean}
+   */
+  static defaultIsEqual(entryA, entryB) {
+    if (entryA instanceof HashEntry) {
+      return entryA.equals(entryB);
+    } else if (entryB instanceof HashEntry) {
+      return false;
+    } else {
+      return super.defaultIsEqual(entryA, entryB);
+    }
+  }
+
+  /**
+   * @param entries {Array<HashEntry | Entry>}
+   * @param [options] {{sortAndExpand?: boolean}}
+   * @returns {FSHashTree}
+   */
+  static fromHashEntries(entries, options = {sortAndExpand: true}) {
```

Comment:
This function is unnecessary, as the `FSTree.fromEntries` correctly handles subclassing:

https://github.com/stefanpenner/fs-tree-diff/blob/0f7eca7d74b34113496dc23d59b45a71b6f61e8a/lib/index.ts#L96

As implemented this will result in the expected subclass:
 `HashTree.fromEntries(/*... */) instanceof HashTree === true`